lib/rubocop/cop/rspec/remove_const.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that `remove_const` is not used in specs.
+      #
+      # @example
+      #   # bad
+      #   it 'does something' do
+      #     Object.send(:remove_const, :SomeConstant)
+      #   end
+      #
```

Comment:
This would only detect top-level const removal.
What if the const is defined in a module/class?
```ruby
class C
  X = 1
end
C.send(:remove_const, :X)
```
Or only the top-level ones are problematic?

Reply:
good point, but the matcher detects this too. At least it did in our codebase and in the adapted test I just pushed :)

Reply:
Nice
The use of `Object` in the node patter in this way was unfamiliar to me.
It seems to work because it checks if the receiver is `=== Object` according to https://docs.rubocop.org/rubocop-ast/node_pattern.html#const-or-const-for-constants

I’m on the fence with this. On the left of the comparison there’ll be a nide with a sting value? Or just a node? Does it mean the cop would also detect `1.send(:remove_const, :Infinity)`?

I’d probably just went with `_` there to avoid confusion